### PR TITLE
Add delete messages API

### PR DESCRIPTION
**Pull request by parfeon**
feat(history): add delete messages API

Add `delete message` API support to the advanced history module.

### CODE REVIEW

File: core/pubnub_advanced_history.c [hunk 1]
```diff
@@ -82,8 +84,8 @@ enum pubnub_res pubnub_message_counts(pubnub_t*   pb,
 }
 
 
-int pubnub_get_chan_msg_counts(pubnub_t* pb, 
-                               size_t* io_count, 
+int pubnub_get_chan_msg_counts(pubnub_t*                     pb,
+                               size_t*                       io_count,
```
Comment (Xavrax) [inline core/pubnub_advanced_history.c:88]: strange formatter ;o

**Review by Xavrax — APPROVED** [review]
seems to be valid
great job!

**Comment by Xavrax** [comment]
@pubnub-release-bot release

File: core/pubnub_advanced_history.c (hunk 1 above)
Comment (parfeon) [reply core/pubnub_advanced_history.c:88]: @Xavrax It aligns argument names, and the next line has a pretty long type, so it added that much to the other lines.
This is the default CLion formatter behavior.

**Pull request merged**